commonmark/src/main/java/org/commonmark/renderer/html/CoreHtmlNodeRenderer.java

```diff
@@ -178,6 +185,9 @@ public void visit(Image image) {
         String altText = altTextVisitor.getAltText();
 
         Map<String, String> attrs = new LinkedHashMap<>();
+        if (context.shouldSanitizeUrls()) {
+            url = context.urlSanitizer().sanitizeImageUrl(url);
```

Comment:
I noticed that for `Links`, you pulled out the `encodeUrl` to happen *after* sanitizing. But here it happens before (expand the context above).

They should be consistent, and I think it makes sense to have `encodeUrl` be after sanitizing.